lib/ronin/recon/builtin/api/hunter_io.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+#
+# ronin-recon - A micro-framework and tool for performing reconnaissance.
+#
+# Copyright (c) 2023-2024 Hal Brodigan (postmodern.mod3@gmail.com)
+#
+# ronin-recon is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ronin-recon is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with ronin-recon.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+require 'ronin/recon/worker'
+
+require 'async/http/internet'
+
+module Ronin
+  module Recon
+    module API
+      #
+      # A recon worker that queries https://api.hunter.io/domain-search
+      # and returns corresponding email addresses.
+      #
+      class HunterIO < Worker
+
+        register 'api/hunter_io'
+
+        summary "Queries the Domains https://api.hunter.io/domain-search"
+        description <<~DESC
+          Queries the Domains https://api.hunter.io/domain-search and returns
+          corresponding email addresses
+        DESC
```

Comment:
Also add documentation here mentioning that `HUNTER_IO_API_KEY` is how to specify the API key.